src/span.js

```diff
@@ -207,23 +207,23 @@ export default class Span {
     /**
      * Adds a log event, or payload to a span.
      *
-     * @param {Object} fields - an object that represents the fields to log.
+     * @param {Object} keyValuePairs - an object that represents the keyValuePairs to log.
      * @param {number} [timestamp] - the starting timestamp of a span.
-     * @param {string} [fields.event] - a string message to be logged on a span.
-     * @param {string} [fields.payload] - an object to be logged on a span as a json string.
      **/
-    log(fields: any, timestamp: ?number): void {
+    log(keyValuePairs: any, timestamp: ?number): void {
         if (this._spanContext.isSampled()) {
-            if (!fields.event && !fields.payload) {
-                this._logger.error('log must be passed either an event of type string, or a payload of type object');
-                return;
-            }
-
-            if (!fields.timestamp) {
-                fields.timestamp = timestamp || Utils.getTimestampMicros();
+            let fields = [];
+            for (let key in keyValuePairs) {
+                let value = keyValuePairs[key];
+                if (keyValuePairs.hasOwnProperty(key)) {
+                    fields.push({'key': key, 'value': value});
+                }
```

Comment:
My preference is to keep tracer and span tags as array({key, value}), rather than a dictionary, because the former allows duplicate keys. However, there are a few places where the API accepts KVs as a map, e.g. options to startSpan, the fields in KV logging, etc. In those places we should call a util function to convert map to an array. But we shouldn't keep repeating the same code over and over.